code/modules/mob/living/living_defense.dm

```diff
@@ -143,26 +136,22 @@
 					src.anchored = 1
 					src.pinned += O
 
-		//Log stuf!
-
+		//Log stuff!
+		if(!src.ckey || src.isDead()) //Message admins if the hit mob is alive and has a ckey
+			return
 		if(!O.fingerprintslast)
 			return
-		var/throwByName = "an unknown inanimate object"
-		if(M)
-			throwByName = M.name
-			M.attack_log += text("\[[time_stamp()]\] <font color='red'>Hit [src.name] ([src.ckey]) with a thrown [O] (speed: [speed])</font>")
-		src.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has been hit with a thrown [O], last touched by [throwByName] ([assailant.ckey]) (speed: [speed])</font>")
-		if(!src.isDead() && src.ckey) //Message admins if the hit mob is alive and has a ckey
-			msg_admin_attack("[src.name] ([src.ckey]) was hit by a thrown [O], last touched by [throwByName] ([assailant.ckey]) (speed: [speed]) (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[src.x];Y=[src.y];Z=[src.z]'>JMP</a>)")
-
-		if(!iscarbon(M))
-			src.LAssailant = null
-		else
-			src.LAssailant = M
-			assaulted_by(M)
+			
+		var/client/assailant = directory[ckey(O.fingerprintslast)]
+		if(assailant && assailant.ckey && assailant.mob)
```

Comment:
it may be useful to have logs even if the assailant doesn't have a ckey, etc (logging a mob throwing an item for example)

Reply:
This is due to gym rats throwing things I think, but I can adjust it

Reply:
of course... while we don't want runtimes, having mob actions that are relevant to players is a good thing (we should probably be logging mob attacks on active players, too... pokemon belt particularly relevant)

Reply:
Actually, if there's no ckey, because fingerprintslast defines the mob that threw it. If there are no fingerprints on the item, there's no mob... Unless objects hold data in addition to fingerprintslast?

Reply:
okay, seems fine then